### DISCUSSION TITLE
Out of 22 clusters, 6 clusters are coming as Macrophages. What do I do?

### DISCUSSION THREAD

**Discussion by shwetajohari** [satijalab/seurat#6471] [Q&A]
Hello, 

My name is Shweta Johari . 
I have four different tumors and I have merged and integrated them, then I got 22 clusters but out of those 22, 5 are coming as macropahes by annotating it using ImmGen. What do I do? 
How does one know that the annotation is correct? 
Please do let me know please. 

Thank you. 

Kind Regards,
Shweta Johari

**Comment by sarakn97** [comment]
Hi Shweta!

I would check the markers for all of the clusters to check if they display macrophage markers. You can use references for cell-type annotation but also must use manual annotation and check marker expression by using FeaturePlot(), FindAllMarkers(), DoHeatmap(), etc....
However, It is very possible that you will have more than one cluster for a cell-type, especially because there are different subsets of macrophages. You just need to manually verify it. I often work with the Immunologists in my lab to help me identify the clusters because they have more knowledge on markers, however, I also have to read through a lot of research papers to find them.

**Comment by shwetajohari** [comment]
Respected Maam,

Thank you. Thank you so much for your reply. I am working on searching the
clusters manually as well as finding different methods to annotate it and
checking the result.
I truly truly want to know if what I am doing is correct? There is no one
here to assist hence I got scared of the results and hence posted.
The code and the findings are correct. Would it please be possible to take
a minute to please check it?
The results and clusters that I can get from this are reliable and
publishable?
Thank you again for replying.

Kind Regards,
Shweta Johari

**Reply by sarakn97** [reply]
I am definitely no expert, I just recently went through the same issue and just spent weeks annotating clusters! For my samples, different from yours, the references were not sufficient and I had issues with incorrect annotations. To manually annotate, I use FeaturePlot() and Heatmaps to identify the main groups : T-Cells, B-Cells, Myeloid, NK Cells and then for example, I take the T-Cell group and re-cluster it so I can find the more defined annotations of T-Cells : CD8+, TREG, CD4+, CD4/8 effector, memory, etc. Your clusters will never be perfect; that is definitely something that I had to realize. I am happy to take a look though!

**Comment by shwetajohari** [comment]
Respected Maam,

Thank you.

Yes, yes. I will be using Find Conserved Markers as well as Heatmap
and Feature Plot.
Thank you so much again.
I have pasted the code that I have used. Is this correct?
Again, Thank you so much for your time.

I have integrated the datasets:
library(Seurat)
library(patchwork)

control = Read10X("C:/Users/TVIT1/Desktop/imtegrationsdj_Tuall/Tu1_099")
inhibitor = Read10X("C:/Users/TVIT1/Desktop/imtegrationsdj_Tuall/Tu2_099")
dnavaccine = Read10X("C:/Users/TVIT1/Desktop/imtegrationsdj_Tuall/Tu3_099")
combitr = Read10X("C:/Users/TVIT1/Desktop/imtegrationsdj_Tuall/Tu4_099")

Created the Seurat Object:
control_seurat = CreateSeuratObject(counts = control$Gene Expression,
project = "control")
inhibitor_seurat = CreateSeuratObject(counts = inhibitor$Gene
Expression, project = "inhibitor")
dnavaccine_seurat = CreateSeuratObject(counts = dnavaccine$Gene
Expression, project = "dnavaccine")
combitr_seurat = CreateSeuratObject(counts = combitr$Gene Expression,
project = "combitr")

Merged the datasets
merge_seurat = merge(x = control_seurat, y = c(inhibitor_seurat,
dnavaccine_seurat, combitr_seurat), add.cell.id = c("control",
"inhibitor","dnavaccine", "combitr"))
***@***.***,5)

Filtering the mitochondrial genes
merge_seurat$mitoRatio <- PercentageFeatureSet(object = merge_seurat,
pattern = "^mt-")
metadata <- ***@***.***
metadata$cells <- rownames(metadata)
library(dplyr)
metadata <- metadata %>%
dplyr::rename(group = orig.ident,
nUMI = nCount_RNA,
nGene = nFeature_RNA)
library(stringr)
metadata$sample <- NA
metadata$sample[which(str_detect(metadata$cells, "^control_"))] <- "control"
metadata$sample[which(str_detect(metadata$cells, "^inhibitor_"))] <- "inhibitor"
metadata$sample[which(str_detect(metadata$cells, "^dnavaccine_"))] <-
"dnavaccine"
metadata$sample[which(str_detect(metadata$cells, "^combitr_"))] <- "combitr"
***@***.*** <- metadata
***@***.***,5)
***@***.***,5)
data.list <- SplitObject(merge_seurat, split.by = "sample")

data.list <- lapply(X = data.list, FUN = function(x) {
x <- NormalizeData(x)
x <- FindVariableFeatures(x, selection.method = "vst")
})

Integration of datasets
features <- SelectIntegrationFeatures(object.list = data.list)
immune.anchors <- FindIntegrationAnchors(object.list = data.list,
anchor.features = features)
immune.combined <- IntegrateData(anchorset = immune.anchors)
DefaultAssay(immune.combined) <- "integrated"
immune.combined <- ScaleData(immune.combined, verbose = FALSE)
immune.combined <- RunPCA(immune.combined, npcs = 30, verbose = FALSE)
immune.combined <- RunUMAP(immune.combined, reduction = "pca", dims = 1:10)
immune.combined <- FindNeighbors(immune.combined, reduction = "pca",
dims = 1:30)
immune.combined <- FindClusters(immune.combined, resolution = 0.5)
p1 <- DimPlot(immune.combined, reduction = "umap", group.by = "sample")
p2 <- DimPlot(immune.combined, reduction = "umap", label = TRUE, repel = TRUE)
DimPlot(immune.combined, reduction = "umap", split.by = "sample")

DefaultAssay(immune.combined) <- "RNA"
library(metap)
all.markers084 <- FindAllMarkers(immune.combined, only.pos = TRUE,
min.pct = 0.25, logfc.threshold = 0.25)

sixteen.markers084 <- FindConservedMarkers(immune.combined, ident.1 =
16, grouping.var = "sample", verbose = FALSE)


Thank you again.

Kind Regards,
Shweta Johari

**Reply by sarakn97** [reply]
Yes, Everything looks correct so far! I think you'll just need further exploration to be sure of your annotations!

Here is an example of what I do:
combined.all is my Seurat Object and I have already performed FindClusters() and visualized them.
```
# Find T-Cell Clusters
FeaturePlot(combined.all, features = c("CD8A", "CD4", "CD8B", "CD40LG", "CD3D", "CD3E"), cols= c("yellow", "blue"), label = TRUE)
# Create Seurat Object of T-Cell Clusters
Tcells <- WhichCells(combined.all, idents= c("3", "0", "5", "7", "10", "9"))

# Perform Sub-clustering on T-Cells
Tcell <- RunPCA(Tcell, npcs = 30, verbose = FALSE)
Tcell <- RunUMAP(Tcell, dims = 1:20) 
Tcell <- FindNeighbors(Tcell, reduction = "pca", dims = 1:20)
Tcell <- FindClusters(Tcell, resolution = 0.6)
Tcell <- SetIdent(Tcell, value = "integrated_snn_res.0.6")

# Visualize Results to find annotations
DimPlot(Tcell, label=TRUE)
DoHeatmap(Tcell, features = c('CD40LG', "CD4","CD8A", "CD8B", "CCR7", "IL7R", "SELL", "DTHD1", "LYST", "LEF1", "CCR4", "GZMH"), assay = "RNA", slot = "data", angle = 90)
```

Also, you may want to play around with the resolution of the DimPlot! This will determine how many clusters you have.

**Comment by shwetajohari** [comment]
Respected Maam,

Thank you. Thank you so much.
For immune cells, I did find another software known as CellKb, try if it
might work for you :)
I will definitely try this. Since I am working on mouse data, I am
searching for genes like CD45  in humans - Ptprc in mice.
Thank you for your time again.
I did play with the resolution from 0.1 to 1 and found 0.5 suitable with 22
clusters.
I was also thinking of checking SCT transform for integrated datasets but I
don't know if it will cause a major difference.
However, Thank you. Thank you for your time.
I really needed someone to just check if what is done is correct or wrong
before I go to my professor with this data.
Thank you again.

Yours sincerely,
Shweta Johari

**Comment by shwetajohari** [comment]
Hello,

I tried to overcome the previous by subsetting it using the following code:
FeaturePlot(immune.combined, features = c("Ptprc", "Cd8a", "Cd4",
"Cd8b", "Cd40lg", "Cd3d", "Cd3e"), cols= c("yellow", "blue"), label =
TRUE)
Tcells <- WhichCells(immune.combined, idents= c("4", "4", "7", "14", "17"))
subsetTcells = subset(immune.combined, cells = Tcells)
subsetTcells = subset(immune.combined, cells = Tcells)
subsetTcells <- FindVariableFeatures(subsetTcells, selection.method =
"vst", nfeatures = 2000)
subsetTcells <- ScaleData(subsetTcells)
subsetTcells <- RunPCA(subsetTcells, features =
VariableFeatures(object = subsetTcells))
DimHeatmap(subsetTcells, dims = 1, cells = 500, balanced = TRUE)
subsetTcells <- RunUMAP(subsetTcells, dims = 1:20)
subsetTcells <- FindNeighbors(subsetTcells, reduction = "pca", dims = 1:20)
subsetTcells <- FindClusters(subsetTcells, resolution = 0.5)
DimPlot(subsetTcells, label=TRUE)
DoHeatmap(subsetTcells, features = c('Cd40lg', "Cd4","Cd8a", "Cd8b",
"Ccr7", "Il7r", "Sell", "Dthd1", "Lyst", "Lef1", "Ccr4", "Gzmh"),
assay = "RNA", slot = "data", angle = 90)


I hope this is right.

Kind Regards,
Shweta Johari

**Comment by shwetajohari** [comment]
Hello,

I tried to subset the T cells and run but I am getting an error.
FeaturePlot(immune.combined, features = c("Ptprc", "Cd8a", "Cd4",
"Cd8b", "Cd40lg", "Cd3d", "Cd3e"), cols= c("yellow", "blue"), label =
TRUE)
Warning message:
In FetchData.Seurat(object = object, vars = c(dims, "ident", features),  :
  The following requested variables were not found: Cd8b

> Tcells <- WhichCells(immune.combined, idents= c("4", "4", "7", "14", "17"))
> Tcell <- RunPCA(Tcell, npcs = 30, verbose = FALSE)
Error in RunPCA(Tcell, npcs = 30, verbose = FALSE) :
  object 'Tcell' not found
> Tcell <- RunPCA(Tcells, npcs = 30, verbose = FALSE)
Error in RowVar(x = object) :
  Not compatible with requested type: [type=character; target=double].
> Tcells <- RunPCA(Tcells, npcs = 30, verbose = FALSE)
Error in RowVar(x = object) :
  Not compatible with requested type: [type=character; target=double].

Is there any particular reason for this?

Thank you.

Kind Regards,
Shweta Johari

**Reply by Deep1707** [reply]
Tcells <- WhichCells(immune.combined, idents= c("4", "4", "7", "14", "17"))
You are  using the same cluster twice.
But the error is lkely because Cd8b is present in your gene list. Try removing it and then run the code.

**Reply by samuel-marsh** [reply]
Hi,

The issue is because the output of WhichCells is not a Seurat object but cell names.  If you want to subset object you should use `subset` and if you are doing it based on active.idents there is no need for WhichCells.

The other errors are due to some lines having TCells and some having TCell.

Best,
Sam

**Comment by shwetajohari** [comment]
Hello,

Thank you so much.
I finally got a trajectory, thank you very much.
However, I do think I am suffering from wrongful annotation and when I
do split var the conserved list, the gene expression for the two
samples is too different.


For example:
DGE All marker
p_val    avg_log2FC    pct.1    pct.2    p_val_adj    cluster    gene
0    1.539682551    0.569    0.048    0    8    Ifng
0    2.024541477    0.63    0.015    0    8    Gzmb
7.86E-190    1.266082397    0.39    0.039    2.54E-185    8    Klrc1
0    2.850705692    0.977    0.05    0    8    Klrk1
0    2.919614667    0.962    0.005    0    8    Ncr1
0    3.531045143    0.992    0.14    0    8    Nkg7

Cluster 8, all these markers are specifically for NK
  you
Now when I do Conserved Markers:

S3                                        S4
p_val    avg_log2FC    pct.1    pct.2    p_val_adj    cluster    gene
              p_val    avg_log2FC    pct.1    pct.2    p_val_adj
cluster    gene
2.63E-13    0.924655229    0.831    0.552    8.48E-09    S3    Zfp36
             0.007857418    0.446668248    0.567    0.455    1    S4
 Rundc3b
1.89E-12    0.813214509    0.894    0.655    6.09E-08    S3    Nfkbia
              1.91E-09    0.39486097    1    1    6.17E-05    S4
Camk1d
3.89E-06    0.75420188    0.831    0.729    0.125717849    S3    Ccl3
              2.07E-15    0.39367236    1    1    6.68E-11    S4
Gm42418
4.41E-05    0.727761135    0.667    0.547    1    S3    Jun
    0.524071972    0.377424289    0.261    0.243    1    S4    Cd74
7.66E-07    0.663527457    0.751    0.626    0.024730895    S3
Dusp1                0.000425095    0.359448725    0.877    0.799    1
   S4    Hist1h1e
5.93E-05    0.586208701    0.529    0.36    1    S3    Fos
   0.00362181    0.34466345    0.305    0.18    1    S4    Hist1h1d
0.076508352    0.573720675    0.413    0.35    1    S3    Wwox
       0.004718349    0.34055597    0.3    0.18    1    S4    Dnah8
1.27E-07    0.511732688    0.931    0.818    0.004089167    S3    Pim1
               0.000840086    0.329477685    0.768    0.704    1    S4
   Ube2i
2.75E-06    0.489834095    0.593    0.404    0.088751858    S3
Kdm6b                0.066639677    0.325250744    0.424    0.354    1
   S4    Klrc1
3.32E-07    0.482781297    0.968    0.98    0.010724135    S3    Sytl3
               0.000550615    0.304928079    0.606    0.439    1    S4
   Wwp2
0.00016038    0.482007123    0.688    0.562    1    S3    Nfkbiz
         0.005950194    0.304078346    0.36    0.249    1    S4
Rbck1
2.03E-05    0.454137589    0.91    0.833    0.654658334    S3    Ier2
              0.124457795    0.281887973    0.261    0.201    1    S4
  Hba-a1
0.003140849    0.452768956    0.603    0.458    1    S3    Xcl1
        5.31E-06    0.277208197    1    1    0.171589164    S4
Lars2
0.000110382    0.450966928    0.847    0.773    1    S3    Sqstm1
          0.012855205    0.272880781    0.621    0.508    1    S4
Mbd5
1.59E-05    0.433487574    0.968    0.921    0.514141466    S3
Tnfaip3                0.167862399    0.263748689    0.557    0.492
1    S4    Atp8a1
3.48E-06    0.426270633    0.937    0.906    0.11219393    S3    Btg2
              0.000398204    0.262360264    0.32    0.159    1    S4
 Sirt2
0.00164783    0.415158087    0.656    0.542    1    S3    Birc3
        0.009263725    0.259288828    0.567    0.455    1    S4
Cct5
0.00054234    0.410198317    0.767    0.67    1    S3    Ppp1r16b
          1.58E-07    0.258135136    1    1    0.005091179    S4
Cdk8

All the genes are different..
So the annotation will be according to the dge or the conserved markers?

Anyway,

Thank you so much for your time.

Kind regards,
Shweta Johari
